### DISCUSSION TITLE
Did the MiddlewareHandlerContext-state change?

### DISCUSSION THREAD

**Discussion by bfkss** [freshframework/fresh#1496] [Q&A] [answered]
I used to assign my state-object directly to `ctx.state` via
```js
ctx.state = {
  foo: 'bar'
}
```
but since 1.2 or 1.3 I can only assign to properties of `ctx.state`, e.g.:
```js
ctx.state.foo = 'bar'
// or
ctx.state.data = {
  foo: 'bar'
}
```

Assigning an object leads to `ctx.state = {}` in the next handler after calling `ctx.next()`.

I haven't seen this documented and am not sure whether it is intended?

**Comment by marvinhagemeister** [comment, marked as answer]
No, this is not intentional and is a bug. Made an issue from that https://github.com/denoland/fresh/issues/1497 .
